### ISSUE TITLE
Search plugin crashes on numeric page titles in `nav`

### ISSUE THREAD

**Issue by planetf1** [squidfunk/mkdocs-material#4817]
### Context

We noticed our github actions which use mkdocs-material failing yesterday, co-inciding with the update of mkdocs-material to 9.0.0 (we do a pip install within our workflow)

### Bug description

mkdocs build is failing with errors such as
```
ERROR    -  Error building page 'release-notes/1-0.md': can only join an iterable
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.1/x64/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/opt/hostedtoolcache/Python/3.11.1/x64/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Link above has the full error

### Related links

* Example action : https://github.com/odpi/egeria-docs/actions/runs/3823032364
* workflow : https://github.com/odpi/egeria-docs/blob/main/.github/workflows/pr-mkdocs.yml

### Reproduction

[example.zip](https://github.com/squidfunk/mkdocs-material/files/10335482/example.zip)

### Steps to reproduce

mkdocs serve

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/bug-report/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) or [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/bug-report/#remove-customizations) before submitting this bug report.
- [X] I have attached a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/bug-report/reproduction/).

**Comment by planetf1** [comment]
The specific cause here seems to be our use of filenames like '1-0.md' where this is also referenced in mkdocs.yml
In 8.5.3 this worked, in 9.0.1 this fails

**Comment by planetf1** [comment]
Ref: original discussion: https://github.com/squidfunk/mkdocs-material/discussions/4816

**Comment by squidfunk** [comment]
Thanks for reporting! I'll look into it.

**Comment by squidfunk** [comment]
Fixed in 4548afb37. Thanks for the excellent reproduction which made fixing this error a breeze. The error was caused because of the use a float in `mkdocs.yml` as a title for a page:

``` yaml
nav:
  ...
  - 1.0: release-notes/1-0.md
```

This was not handled by the search plugin before. To mitigate the issue before the next release (we _just_ pushed out 9.0.1 and want to collect some more bug fixes before 9.0.2), put `1.0` in quotes: `"1.0"` or [install from git](https://squidfunk.github.io/mkdocs-material/getting-started/#with-git).

**Comment by planetf1** [comment]
Brilliant. Thanks for super quick fix and glad we could help improve! Mkdocs material is great!

**Comment by squidfunk** [comment]
Released as part of [9.0.2](https://github.com/squidfunk/mkdocs-material/releases/tag/9.0.2).
